samples/subsys/mgmt/mcumgr/smp_svr/prj_udp.conf

```diff
@@ -0,0 +1,40 @@
+# Enable mcumgr.
+CONFIG_MCUMGR=y
+
+# Some command handlers require a large stack.
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+
+# Ensure an MCUboot-compatible binary is generated.
+CONFIG_BOOTLOADER_MCUBOOT=y
+
+# Enable the UDP mcumgr transports.
+CONFIG_MCUMGR_SMP_UDP=y
+CONFIG_NET_BUF_USER_DATA_SIZE=8
+
+# Network settings
+CONFIG_NETWORKING=y
+CONFIG_NET_MGMT=y
+CONFIG_NET_L2_ETHERNET_MGMT=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_UDP=y
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_AUTO_INIT=y
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="169.254.0.100"
```

Comment:
Why such hard coded IP address? For IPv4, it is also possible to enable CONFIG_NET_IPV4_AUTO which enables IPv4 auto address configuration (RFC 3927)

Reply:
Honestly, I haven't been able to make CONFIG_NET_IPV4_AUTO work properly for me, so I opted for hardcoded addresses instead. I'm using the same adresses as the echo_client socket example.

Reply:
> I haven't been able to make CONFIG_NET_IPV4_AUTO work properly for me,

Ok, it would be nice to know why IPv4 auto addresses are not working. Could you create an issue for this one so we can try to figure out if there is a real bug in it or if it is just a configuration issue.

Reply:
I just tested again and it does in fact work. I had some other issues earlier that I attributed to CONFIG_NET_IPV4_AUTO.

But, I'm still not convinced using automatic ip configuration is a good fit for this sample. The mcumgr tool needs to know the ip address of the device, so the sample needs to print that address and the user needs to copy that every time. And what about ipv6?

So, I suggest we stick with static addresses for this sample.

Reply:
Sure, using static addresses is ok. I was just wondering why you have the address selected from 169.254/16 subnet which is allocated for IPv4 auto addresses.